esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2944,21 +2959,27 @@ bool CFileSprayEx::checkDropZoneIPAndPath(double clientVersion, const char* drop
     if (isEmptyString(netAddr) || isEmptyString(path))
         throw MakeStringException(ECLWATCH_INVALID_INPUT, "NetworkAddress or Path not defined.");
 
-    Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory(true);
-    Owned<IConstEnvironment> constEnv = envFactory->openEnvironment();
-    Owned<IConstDropZoneInfoIterator> dropZoneItr = constEnv->getDropZoneIteratorByAddress(netAddr);
-    ForEach(*dropZoneItr)
+    ForEachItemIn(i, allTpDropZones)
     {
-        SCMStringBuffer directory, name;
-        IConstDropZoneInfo& dropZoneInfo = dropZoneItr->query();
-        dropZoneInfo.getDirectory(directory);
-        if (directory.length() && (strnicmp(path, directory.str(), directory.length()) == 0))
-        {
-            if (isEmptyString(dropZoneName))
-                return true;
+        IConstTpDropZone& dropZone = allTpDropZones.item(i);
+        const char* name = dropZone.getName();
+        if (isEmptyString(name))
+            continue;
+        if (!isEmptyString(dropZoneName) && !streq(dropZoneName, name))
+            continue;
 
-            dropZoneInfo.getName(name);
-            if (strieq(name.str(), dropZoneName))
+        const char* prefix = dropZone.getPath();
+        if (isEmptyString(prefix))
+            continue;
+        if (strnicmp(path, prefix, strlen(prefix)) != 0)
+            continue;
+
+        IArrayOf<IConstTpMachine>& tpMachines = dropZone.getTpMachines();
+        ForEachItemIn(ii, tpMachines)
+        {
+            IConstTpMachine& tpMachine = tpMachines.item(ii);
+            const char* netAddress = tpMachines.item(ii).getNetaddress();
+            if (streq(netAddr, netAddress))
```

Comment:
I think in the cloud the user should be dealing with DNS names rather than IPs.  Can we make sure if they are providing DNS name we match that?  We can still let them provide an IP for now just in case, but we should probably frown on it and only be using DNS names.  Unless I'm missing something.